crates/ra_assists/src/flip_binexpr.rs

```diff
@@ -0,0 +1,149 @@
+use hir::db::HirDatabase;
+use ra_syntax::ast::{AstNode, BinExpr, BinOp};
+
+use crate::{AssistCtx, Assist, AssistId};
+
+/// Flip binary comparison expressions (==, !=, >, >=, <, <=).
+pub(crate) fn flip_binexpr(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let expr = ctx.node_at_offset::<BinExpr>()?;
+    let lhs = expr.lhs()?.syntax();
+    let rhs = expr.rhs()?.syntax();
+    let op_range = expr.op()?.range();
+    // The assist should be available only if the cursor is on the operator
+    let cursor_in_range = ctx.frange.range.is_subrange(&op_range);
+    // The assist should be available only for these binary operators
+    // (it should not change the meaning of the expression)
+    let allowed_ops = [
+        BinOp::EqualityTest,
+        BinOp::NegatedEqualityTest,
+        BinOp::GreaterTest,
+        BinOp::GreaterEqualTest,
+        BinOp::LesserTest,
+        BinOp::LesserEqualTest,
+    ];
+    let op_kind = expr.op_kind()?;
+    if !cursor_in_range || !allowed_ops.iter().any(|o| *o == op_kind) {
+        return None;
+    }
+    let new_op = match op_kind {
+        BinOp::GreaterTest => Some("<"),
+        BinOp::GreaterEqualTest => Some("<="),
+        BinOp::LesserTest => Some(">"),
+        BinOp::LesserEqualTest => Some(">="),
```

Comment:
There's some duplication with `allowed_ops`, perhaps we need just a single `flip_op -> Option<str>` function?

Reply:
The first check disables the assist while this one avoids replacing the operator when it's not needed (for example there is no need to change `==` but `<` should be changed to `>`).

Reply:
I think it's simpler to just change `==` to `==` :)

Reply:
It's slower! :D BTW if we allow (almost) every binary operator... only a few will need to be replaced

Reply:
Slowness should not matter here. But yeah, only a few operators need a flip... Anyway, don't have a strong opinion here